### DISCUSSION TITLE
Linting clarification

### DISCUSSION THREAD

**Discussion by priyankavit** [stackrox/kube-linter#511] [Q&A]
Can I lint only pull request or changes using kubelinter instead of linting whole kubernetes folder everytime?

**Comment by Zaijo** [comment]
Hello, what is you motivation for that @priyankavit ? Does `kube-linter lint` run for too long?

**Reply by priyankavit** [reply]
Thank you @Zaijo for replying , Actually Motivation for linting only changes in kubernetes files because the folder containing around 40+ kubernetes yaml files gets linted and takes more time. So I wanted to lint only the file where I makes changes not all files everytime.

**Reply by Zaijo** [reply]
AFAIK using GitLab pipelines you can create conditional jobs for MR running only when certain files are updated. Don't know how GitHub or your pipeline implementation handles this. 

So what you can do is split your manifests by intention or by customer and then lint these groups separately in independent runs based on modified files.

See https://docs.gitlab.com/ee/ci/yaml/#ruleschanges

**Reply by priyankavit** [reply]
Thank you @Zaijo This will be helpful.

**Comment by janisz** [comment]
If you want to lint only changed files then this could be helpful https://gist.github.com/seeliang/0f0de424d1cdc4541c338f4ee93b7e6a#for-stylelint it's for stylelint but similar appraoch should work for kube-linter too.

**Comment by priyankavit** [comment]
Thank you for response. I tried this and able to lint only the changed files now by cloning MR branch and then ran below command

docker run -v $WORKSPACE:/dir -v $WORKSPACE:/gitDiff -w /dir stackrox/kube-linter lint /gitDiff/$(git diff --name-only HEAD^ HEAD)
